tools/site_cobble/git_revision.py

```diff
@@ -0,0 +1,471 @@
+# This is python port of https://github.com/passsy/git-revision/
+# with some modifications to our preferences and use-cases.
+# See LICENSE.git_revision.md in the root of the repository.
+# -------------------------------------------------------------
+#
+#   Copyright 2021 Oxide Computer Company
+#   Copyright 2018 Pascal Welsch
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+import argparse
+import datetime
+import pathlib
+import re
+import subprocess
+
+from functools import cached_property, cache
+from typing import List, Iterator, Optional, Union, Tuple
```

Comment:
Question (no action needed):
Is there any difference between using the `typing` module's `List` versus just the built-in `list`? My working understanding is that `List` is just an alias.

Reply:
I think at this point there isn't much difference, but that hasn't always been true in py3 iirc and I was doing type annotations before they allowed the list aliasing.  I did pay around in a local copy and it looks like you can indeed swap list for typying.List. If used for function annotation the syntax still requires square brackets for the generics.

Generally once I have to import typing to get other things that aren't built-in I just end up doing that.